mindsdb/integrations/handlers/slack_handler/slack_handler.py

```diff
@@ -394,14 +399,21 @@ def _process_websocket_message(client: SocketModeClient, request: SocketModeRequ
                 return
 
             payload_event = request.payload['event']
-            if payload_event['type'] not in ('message', 'app_mention'):
+
+            if payload_event['type'] == 'message' and payload_event['channel_type'] != 'im':
```

Comment:
[Here](https://github.com/mindsdb/mindsdb/pull/9402/files#diff-a28ff1351772fd317f3255e86983b784fb17ebb20c90f5cd4b63637627c5e931L378) I removed `['channel_type'] != 'im'` because someone wanted to get answers in channels...

Reply:
@ea-rus Without this condition, the bot will respond to all messages posted in channels. Even when it is not directed at the bot. For instance, even if it mentions someone else using '@'.

Reply:
I think there are two situations that our chat-bots should post responses (I can confirm that this is what Sema4 wants as well),
1. When a direct message is sent to the installed app.
2. When the app is mentioned within a message in a channel.

@ea-rus @ZoranPandovski What do you guys think?

Reply:
@ea-rus @ZoranPandovski I think that if we are in agreement, we can remove the message_channels event subscription from this list,
![image](https://github.com/user-attachments/assets/7d983d4c-18bc-4271-8fa3-f0935155845a)

The bot will then only respond to direct messages to the app or when it is explicitly mentioned. What do you think?

Reply:
looks good. in that case to answer in channels or not will be managed by bot permissions, right?

Reply:
Yes, if we remove this permission, no event will be triggered when normal messages are posted in channels; only when the app is mentioned.

One more thing; I think since we are caching the channels in the code, we might need to update the list anytime an operation is performed on channels such as creating new one or updating an existing one. To do this, we would need to be subscribe to additional events; probably all of these:
![image](https://github.com/user-attachments/assets/0bc204af-7d46-44ab-b47f-a9d3aa665f32)

What is your opinion on this?